Gems/Terrain/Code/Source/TerrainRenderer/TerrainMacroMaterialManager.cpp

```diff
@@ -226,26 +210,25 @@ namespace Terrain
             "Changed announcements for materials that haven't had a OnCreated event sent, or the terrain feature processor isn't properly tracking macro materials.");
         
         MacroMaterial& macroMaterial = m_macroMaterials[entityId];
-        macroMaterial.m_data.m_bounds = newRegion;
+        macroMaterial.m_data.m_minBounds = AZ::Vector2(newRegion.GetMin());
+        macroMaterial.m_data.m_maxBounds = AZ::Vector2(newRegion.GetMax());
 
         UpdateMacroMaterialShaderEntry(m_materialShaderData.GetElement(macroMaterial.m_materialRef), macroMaterial);
 
         AZ::Aabb changedRegion = oldRegion;
         changedRegion.AddAabb(newRegion);
 
-        ForMacroMaterialsInBounds(changedRegion,
-            [&](uint32_t idx, const AZ::Vector2& corner)
+        ForMacroMaterialsInBounds(AZ::Vector2(changedRegion.GetMin()), AZ::Vector2(changedRegion.GetMax()),
+            [&](uint32_t idx, const AZ::Vector2& tileMin)
             {
-                AZ::Aabb tileAabb = AZ::Aabb::CreateFromMinMaxValues(
-                    corner.GetX(), corner.GetY(), m_terrainBounds.GetMin().GetZ(),
-                    corner.GetX() + MacroMaterialGridSize, corner.GetY() + MacroMaterialGridSize, m_terrainBounds.GetMax().GetZ());
-
-                bool overlapsNew = tileAabb.Overlaps(newRegion);
+                AZ::Vector2 tileMax = tileMin + AZ::Vector2(MacroMaterialGridSize);
+                bool overlapsNew =
+                    tileMin.IsLessEqualThan(macroMaterial.m_data.m_maxBounds) &&
+                    tileMax.IsGreaterEqualThan(macroMaterial.m_data.m_minBounds);
```

Comment:
If you intentionally aren't using <= / >= above, then this should probably change?

Reply:
yeah this should be strict I think since it's used to figure out which tiles need which macro materials. nice catch.